From 3a2ac517839968d685a253a9af5a7cd26d1fdb4d Mon Sep 17 00:00:00 2001
From: Daniel Budris <budris@amazon.com>
Date: Fri, 23 Sep 2022 09:26:30 -0400
Subject: [PATCH] skip privilleged syscall tests in fargate execution context

Fargate execution context does not support privilleged execution of commands like `unshare`.
The syscall tests that depend on such calls are already configured to skip if the env is docker or lxc
This patch updates the check to include Fargate

---
 src/syscall/exec_linux_test.go | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/syscall/exec_linux_test.go b/src/syscall/exec_linux_test.go
index df4ed62630..74acb6a277 100644
--- a/src/syscall/exec_linux_test.go
+++ b/src/syscall/exec_linux_test.go
@@ -33,6 +33,10 @@ func isLXC() bool {
 	return os.Getenv("container") == "lxc"
 }
 
+func isFargate() bool {
+	return os.Getenv("AWS_EXECUTION_ENV") == "AWS_ECS_FARGATE"
+}
+
 func skipInContainer(t *testing.T) {
 	// TODO: the callers of this func are using this func to skip
 	// tests when running as some sort of "fake root" that's uid 0
@@ -42,6 +46,9 @@ func skipInContainer(t *testing.T) {
 	// for what we need exactly (which capabilities are active?),
 	// instead of just assuming "docker == bad". Then we'd get more test
 	// coverage on a bunch of builders too.
+	if isFargate() {
+		t.Skip("skip this test in Fargate execution context")
+	}
 	if isDocker() {
 		t.Skip("skip this test in Docker container")
 	}
-- 
2.30.1 (Apple Git-130)

